lstchain/reco/r0_to_dl1.py

```diff
@@ -838,3 +669,229 @@ def rescale_dl1_charge(event, scaling_factor):
 
     for tel_id, tel in event.dl1.tel.items():
         tel.image *= scaling_factor
+
+
+def event_timing_pointing(dl1_container,
+                          config,
+                          event,
+                          telescope_id,
+                          ucts_t0_dragon=math.nan,
+                          dragon_counter0=math.nan,
+                          ucts_t0_tib=math.nan,
+                          tib_counter0=math.nan,
+                          first_valid_ucts=None,
+                          first_valid_ucts_tib=None,
+                          previous_ucts_time_unix=[],
+                          previous_ucts_trigger_type=[],
+                          pointings=None,
+                          drive_data=None,
+                          ):
+    """
+    Compute the correct event timestamp and pointing
+
+    Parameters
+    ----------
+    dl1_container: `lstchain.io.lstcontainer.DL1ParametersContainer`
+    config: dict
+    event: `ctapipe.containers.DataContainer`
+    telescope_id: int
+    ucts_t0_dragon: float
+    dragon_counter0: float
+    ucts_t0_tib: float
+    tib_counter0: float
+    first_valid_ucts: float or None
+    first_valid_ucts_tib: float or None
+    previous_ucts_time_unix: list
+    previous_ucts_trigger_type: list
+    pointings: None or `lstchain.pointing.PointingPosition`
+    drive_data: None or dict
+
+    Returns
+    -------
+    first_valid_ucts, first_valid_ucts_tib, previous_ucts_time_unix, previous_ucts_trigger_type, dragon_time
+    """
+    # GPS + WRS + UCTS is now working in its nominal configuration.
+    # These TS are stored into ucts_time container.
+    # TS can be alternatively calculated from the TIB and
+    # Dragon modules counters based on the first valid UCTS TS
+    # as the reference point. For the time being, the three TS
+    # are stored in the DL1 files for checking purposes.
+
+    module_id = 82  # Get counters from the central Dragon module
+
+    if math.isnan(ucts_t0_dragon) and math.isnan(dragon_counter0) \
+            and math.isnan(ucts_t0_tib) and math.isnan(tib_counter0):
+        # Dragon/TIB timestamps not based on a valid absolute reference timestamp
+
+        dragon_time = (
+                event.lst.tel[telescope_id].svc.date +
+                event.lst.tel[telescope_id].evt.pps_counter[module_id] +
```

Comment:
I recently learned that the LST data comes in random module order and currently I think only the `waveform` is reordered by the event source to actual pixel id order.

I suspect that here you are not really looking at module 82 but at the random module that is in position 82.

@FrancaCassol, can you confirm?

Reply:
I confirm @maxnoe, 
I think I already corrected this in a mail exchange, but without too much success ;-)
There are two errors here: the module id order is not always following the array ranking order (even if sometime it does) AND the central module id is  not 82 but 132.
To make things correct one should use the
central_module_rank=np.where(event.lst.tel[tel_id].svc.module_ids==132)

Reply:
Do you know whether this has any consequence for the last reprocessing? @morcuended : is the counter value provided as input from Isidro's daily check (truly) coming from the central module?

Reply:
In theory, it should not really matter which dragon board you take as reference, right?

Reply:
I would say it does not really matter as @maxnoe says since they are pretty well aligned. Let me check it with Isidro.

Reply:
Isidro explained to me how he was choosing these modules:

There are the `expected_modules_id` and then `expected_pixels_id` arrays. In order to get the counters corresponding to the central module (#132), he looks for 132 in the `expected_modules_id` (as Franca wrote above). That value seemed to be 82 in the LST-CAM2 for last months (i.e. `expected_modules_id[82]=132`). Hence counters are extracted according to `expected_modules_id` (`counters[82, pps_index]` and `counters[82. 10MHz_index]`). This is why I put that 82 number in the code. Checking the number of pixels corresponding to that 82 indexes, one should obtain pixels 1 to 6 (the ones of the central module).

In any case, all modules are well synchronized. According to Isidro,  PPS counters are well aligned and it may happen that some of the 10MHz counters are shifted by 100 nsec in the worst case.

Reply:
My understanding is, that the order is random, per event. It might be that for some files, due to network latency, module 132 was often or even always at position 82. But from what Franca told me, this is nothing that should be relied on.

Reply:
The `expected_modules_id` is given with every even for a reason.

Reply:
Yes, for sure we have to use what Franca suggested. I was just trying to understand why I put that value there.